index.ts

```diff
@@ -76,7 +104,9 @@ function writeProtocolData (writer: Writer, protocolData: ProtocolData[]) {
   writer.writeUInt(lengthPrefix, lengthPrefixLengthPrefix)
 
   for (const p of protocolData) {
-    writer.writeVarOctetString(Buffer.from(p.protocolName, 'ascii'))
+    writer.writeVarOctetString(
+      protocolNameCache[p.protocolName] ||
```

Comment:
we should allow the cache to be passed in as another optional argument, that way if someone writes a plugin that uses new protocol names they can pass them into here instead of patching the library. I think it's a good idea to still keep the most common ones in here, though.